File: Examples/ObjectiveCExample/BRG-ObjectiveCExample/ViewController.m
//
//  ViewController.m
//  BRG-ObjectiveCExample
//
//  Created by Nguyễn Mai Quân on 23/9/25.
//

#import "ViewController.h"
#import <WebKit/WebKit.h>
#import <BridgewellEventSDK/BridgewellEventSDK-Swift.h>

@interface ViewController ()
@property (weak, nonatomic) IBOutlet WKWebView *webView;
@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];

    // Initialize the SDK
    BridgewellConfig *config = [[BridgewellConfig alloc] initWithAppIdOverride:nil
                                                               loggingEnabled:YES];
    [[BridgewellEvent shared] initializeWithConfig:config];

    // Load your web content
    NSURL *url = [NSURL URLWithString:@"https://img.scupio.com/cat/webview-test.html"];
    NSURLRequest *request = [NSURLRequest requestWithURL:url];
    [self.webView loadRequest:request];

    // Register WebView for comprehensive data injection (includes device info)
    // This method will inject full device data including BwsDevice, BwsGeo, BwsMobile, etc.
    // Location services are enabled by default in the enhanced SDK
    [[BridgewellEvent shared] registerContentWebViewWithAdInfo:self.webView];
}


@end

File: Examples/SwiftExample/BRG-SwiftExample/ViewController.swift
//
//  ViewController.swift
//  BRG-SwiftExample
//
//  Created by Nguyễn Mai Quân on 23/9/25.
//

import UIKit
import WebKit
import BridgewellEventSDK

class ViewController: UIViewController {

    @IBOutlet weak var webView: WKWebView!

	override func viewDidLoad() {
		super.viewDidLoad()

        let config = BridgewellConfig(
            appIdOverride: nil,
            loggingEnabled: true
        )
        BridgewellEvent.shared.initialize(config: config)

        BridgewellEvent.shared.registerContentWebViewWithAdInfo(webView!)
	}

    override func viewDidAppear(_ animated: Bool) {
        super.viewDidAppear(animated)

        DispatchQueue.main.asyncAfter(deadline: .now() + 0.5) {
            self.handleLoadRemoteWebview()
        }
    }

    private func handleLoadWebview() {
        if let path = Bundle.main.path(forResource: "test-device-injection", ofType: "html"),
           let htmlString = try? String(contentsOfFile: path, encoding: .utf8) {
            let baseURL = URL(fileURLWithPath: path)
            webView.loadHTMLString(htmlString, baseURL: baseURL)
        } else {
            if let url = URL(string: "https://img.scupio.com/cat/webview-test.html") {
                webView.load(URLRequest(url: url))
            }
        }
    }

    private func handleLoadRemoteWebview() {
        if let url = URL(string: "https://img.scupio.com/cat/webview-test.html") {
            webView.load(URLRequest(url: url))
        }
    }
}

